core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -117,11 +117,28 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.DELETE_RECORDS => handleDeleteRecordsRequest(request)
         case ApiKeys.INIT_PRODUCER_ID => handleInitProducerIdRequest(request)
         case ApiKeys.OFFSET_FOR_LEADER_EPOCH => handleOffsetForLeaderEpochRequest(request)
-        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleAddPartitionToTxnRequest(request)
-        case ApiKeys.ADD_OFFSETS_TO_TXN => handleAddOffsetsToTxnRequest(request)
-        case ApiKeys.END_TXN => handleEndTxnRequest(request)
-        case ApiKeys.WRITE_TXN_MARKERS => handleWriteTxnMarkersRequest(request)
-        case ApiKeys.TXN_OFFSET_COMMIT => handleTxnOffsetCommitRequest(request)
+        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleWithInterBrokerVersionAtLeast(KAFKA_0_11_0_IV0, request, (error:Errors) => {
```

Comment:
I think we could turn this into a simple assertion which could be added to the respective handlers so we don't make this block messy. Something like this:

```scala
def ensureInterBrokerVersion(version: Short) {
  if (config.interBrokerProtocolVersion < version)
    throw new InvalidRequestException()
}
```
The exception will be caught and passed to `getErrorResponse`, so the effect should be the same.

Also, I'm debating whether InvalidRequest is what we want. We could use UnsupportedVersion also, but it's a bit odd that we raise this after we have already advertised support in the ApiVersions response. Perhaps that means we shouldn't even advertise them until in ApiVersions until the inter-broker protocol has been updated?

cc @ijuma